src/defaultsettings.cpp

```diff
@@ -26,9 +26,73 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "mapgen/mapgen.h" // Mapgen::setDefaultSettings
 #include "util/string.h"
 
+
+/*
+ * inspired by https://github.com/systemd/systemd/blob/7aed43437175623e0f3ae8b071bbc500c13ce893/src/hostname/hostnamed.c#L406
+ * this could be done in future with D-Bus using query:
+ * busctl get-property org.freedesktop.hostname1 /org/freedesktop/hostname1 org.freedesktop.hostname1 Chassis
+ */
+bool detect_touch()
+{
+#if defined(__ANDROID__)
+	return true;
+#else
+	std::string chassis_type;
+#if defined(__linux__)
```

Comment:
`chassis_type` is only used by the Linux code, so it can be inside `#if defined(__linux__)`

Reply:
Done, I did small restructuring in the fixup :)